src/utils/fs.ts

```diff
@@ -50,6 +50,34 @@ export function isFile (path: string): Promise<boolean> {
 }
 
 /**
+ * Verify a path exists and is a directory.
+ */
+export function isDirectory (path: string): Promise<boolean> {
+  return stat(path).then(stat => stat.isDirectory(), () => false)
+}
+
+function filterAsync<T>(items: T[], predicate: { (item: T): Promise<boolean> }): Promise<T[]> {
+  const pendingFilters = items.map(item => predicate(item))
+
+  return Promise.all(pendingFilters)
+    .then(filters => {
+      return items.filter((item: T, index: number) => filters[index])
+    })
+}
+
+/**
+ * Read the sub directories under in a path.
+ */
+export function readSubDirs (path: string): Promise<string[]> {
```

Comment:
Is there a need for this? I was thinking you could just use `main.d.ts` as the source of truth and start wiping extra dependencies from there. You could stat each dependency before you rimraf for emit, but I didn't think there was much more than that.